### PR TITLE
Implementation of multiple results per match.

### PR DESCRIPTION
**Pull request by citron0xa9**

### CODE REVIEW

File: spec/models/game_result_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe GameResult, type: :model do
+  it "is valid when produced by a factory" do
+    result = FactoryBot.build(:game_result)
+    result.save
```
Comment (nikriek) [inline spec/models/game_result_spec.rb]: You dont need to save the result. Make it a one line liek

```
expect(FactoryBot.build(:game_result)).not_to be_valid
```

File: spec/models/game_result_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    expect(result).to be_valid
+  end
+
+  it "fails validation with non-number scores" do
+    result = FactoryBot.build(:game_result, score_home: "2a")
```
Comment (nikriek) [inline spec/models/game_result_spec.rb]: You might also check for the existence of a `score_home` in the errors dict after calling `is_valid?`

```ruby
r = FactoryBot.build(:game_result, score_home: "2a")
expect {r.is_valid?}.to change { r.errors }.from([]).to([score_home: [...]])
# OR
expect {r.is_valid?}.to change { r.errors[:score_home].count }.from(0).to(1)
```

File: spec/models/game_result_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    result.save
+
+    expect(result).not_to be_valid
+  end
+
+  it "is valid without any scores given" do
+    result = GameResult.new
+    match = FactoryBot.create(:match)
```
Comment (nikriek) [inline spec/models/game_result_spec.rb]: just pass it directly in the constructor and check for validity

File: spec/factories/matches.rb [hunk 4]
```diff
@@ -22,10 +22,15 @@
     association :team_away, factory: :team
     association :team_home, factory: :team
     association :event, factory: :event
-    score_home { rand(10..20) }
-    score_away { rand(1..9) }
     gameday 0
     points_away 3
     points_home 1
+
+    factory :match_with_results do
```
Comment (nikriek) [inline spec/factories/matches.rb]: Have a look at rspec traits

**Review by nikriek — APPROVED** [review]
Looks good. just minor stuff

File: spec/factories/matches.rb (hunk 4 above)
Comment (citron0xa9) [reply spec/factories/matches.rb]: I think this is not doable because the callback `after(:create)` is used

**Pull request merged**